modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/service/impl/BlogsEntryLocalServiceImpl.java

```diff
@@ -1467,6 +1467,22 @@ public BlogsEntry updateStatus(
 				}
 			}
 
+			if (GetterUtil.getBoolean(
+				serviceContext.getAttribute("addEntryResources"))) {
+
+				if (serviceContext.isAddGroupPermissions() ||
+					serviceContext.isAddGuestPermissions()) {
+
+					addEntryResources(
+						entry, serviceContext.isAddGroupPermissions(),
+						serviceContext.isAddGuestPermissions());
+				}
+				else {
+					addEntryResources(
+						entry, serviceContext.getModelPermissions());
+				}
+			}
+
```

Comment:
@kristofloneck The code for both conditions is identical. Can you merge both conditions?
```
		if ((oldStatus == WorkflowConstants.STATUS_DRAFT) ||
                     GetterUtil.getBoolean(
				serviceContext.getAttribute("addEntryResources"))) {

				if (serviceContext.isAddGroupPermissions() ||
					serviceContext.isAddGuestPermissions()) {

					addEntryResources(
						entry, serviceContext.isAddGroupPermissions(),
						serviceContext.isAddGuestPermissions());
				}
				else {
					addEntryResources(
						entry, serviceContext.getModelPermissions());
				}
			}
```

Reply:
fixed